### ISSUE TITLE
Feature request: add posibility to enable lumunarr only for specific library instead of media type

### ISSUE THREAD

**Issue by Raiiyzen** [chadwpalm/Lumunarr#25]
Actually whe only have the possibility to enable lumunarr for media types (Movies, TV Shows, trailer / pre roll).
It would be nice if we can enable lumunarr scenes with specific library instead of media types. 

For example in my purpose, i would like to enable this only for my 4K movie library, because i've only enabled credit detections for that specific library.

**Comment by chadwpalm** [comment]
Given that movie libraries can only be of one type anyway, it really does make more sense that instead of choosing the media type you would choose the library instead, and also keep the All option there.

Up until know I've limited all Plex API calls to the user's Plex.tv account, because that's all I had ever needed. No API calls directly to the PMS are done, so there currently isn't a way to grab the list of libraries. This will need to change.

It would add a dimension to Lumunarr which I should have done from the beginning and that is being able to choose which Plex Server you want to connect Lumunarr to because your account could technically be connected to more than one server. It also would allow external IP access if Lumunarr was running from a different location than the PMS.

I'm currently working on Hue Cloud API integration, but this could be something I can also work on in parallel for a massive 1.5.0 release.

**Comment by chadwpalm** [comment]
I took some time to look into this further, and as it turns out the webhooks that are sent from the Plex server do not contain information on which library the media comes from, only the library type.

At this point you would need to petition with Plex to add that information into the webhook metadata and then I'd be able to use it to do what you are looking for.

**Comment by Raiiyzen** [comment]
Hello,

Can confirm that it works great, thank you ! 

however it seems like scrobble delay doesn't work for me, my lights powering up a bit sooner, but i can open an another issue if you want.

**Comment by chadwpalm** [comment]
Let me take a look at it. If it looks like an easy fix then I'll go ahead and do that. No need to open up a new issue.

**Comment by chadwpalm** [comment]
I ran through a series of tests on two different devices. My Roku, and the Windows desktop client.

I can confirm through the logs that the scrobble delay is working correctly.

Roku:
No delay:
[2025-04-12 21:45:23] [info] Waiting 0ms before recalling scene
[2025-04-12 21:45:23] [info] Scrobble scene was recalled All on Chad's Roku Ultra (Roku)
5 second delay:
[2025-04-12 21:46:15] [info] Waiting 5000ms before recalling scene
[2025-04-12 21:46:20] [info] Scrobble scene was recalled All on Chad's Roku Ultra (Roku)
4 second delay:
[2025-04-12 22:15:22] [info] Waiting 4000ms before recalling scene
[2025-04-12 22:15:26] [info] Scrobble scene was recalled All on Chad's Roku Ultra (Roku)

Windows:
5 second delay:
[2025-04-13 08:23:32] [info] Waiting 5000ms before recalling scene
[2025-04-13 08:23:37] [info] Scrobble scene was recalled movie on NEW-MONSTER (windows)
3 second delay:
[2025-04-13 08:24:42] [info] Waiting 3000ms before recalling scene
[2025-04-13 08:24:45] [info] Scrobble scene was recalled movie on NEW-MONSTER (windows)

The first line of each instance occurs when the webhook comes in, and the second line is after the delay and when the lights are triggered. The timestamps seem to confirm that the delay is working correctly.

I think the problem is in the erratic nature of when the scrobble webhook comes in. On my Roku it was consistently triggering at the exact same media timestamp, but on my Windows client it was triggering at various timestamps (sometimes up to a 5 second variance). In fact, on Windows there were some instances when the webhook didn't trigger at all unless I went more than 10 seconds before that point, and in one case it wouldn't trigger at all unless I completely stopped the video and started it again. This wasn't a case of Lumunarr missing the hooks, but Plex just not sending them at all. I know this because I was running a parallel webhook to a receiving website that can also capture webhooks. It was definitely on the Plex server side.

I have a suspicion that a lot of this could be based on the intervals of how each client reports it's state back to the server and how each client was designed to do it. I just don't think this is a Lumunarr issue. I can confirm all webhooks that are being sent are being captured and the delay is working as it should.

**Comment by Raiiyzen** [comment]
Thank you for your answer ! 

If scrobble delay behavior differs depending on device, could it be possible to set a scrobble value higher than the actual max (5s) ? 

Cheers

**Comment by chadwpalm** [comment]
> If scrobble delay behavior differs depending on device, could it be possible to set a scrobble value higher than the actual max (5s) 

Yeah, I can bundle the change into the next release.

**Comment by Raiiyzen** [comment]
hi, any ETA for the next release ? 

Cheers

**Comment by Raiiyzen** [comment]
With the actual settings, my lights are triggered 5 seconds too early, even with the delay set to 5 seconds, so for my case only, a delay of 10 seconds would be good. But I think other people would be interested in delaying more than 10 seconds, so maybe 30 seconds would be enough for most cases?

**Comment by Raiiyzen** [comment]
If you have discord, I can sent to you a video I think it would be better

**Comment by chadwpalm** [comment]
Going to close this issue as I believe it is completed. If still having problems please open a new issue.
